File: src/net/auoeke/cin/lexer/lexeme/CommentLexeme.java
package net.auoeke.cin.lexer.lexeme;

import net.auoeke.cin.lexer.error.SyntaxError;

public final class CommentLexeme extends Lexeme {
    public final String value;
    public final Type type;

    public CommentLexeme(int line, int column, Type type, String value, SyntaxError error) {
        super(line, column, error);

        this.type = type;
        this.value = value;
    }

    public CommentLexeme(int line, int column, Type type, String value) {
        this(line, column, type, value, null);
    }

    @Override public Type type() {
        return this.type;
    }

    @Override public String toString() {
        return switch (this.type()) {
            case LINE_COMMENT -> "##$value";
            case BLOCK_COMMENT -> "/*$value*/";
            default -> throw new IllegalStateException(this.type.toString());
        };
    }
}

File: src/net/auoeke/cin/lexer/lexeme/CommentLexeme.kt
package net.auoeke.cin.lexer.lexeme

import net.auoeke.extensions.*

class CommentLexeme(line: Int, column: Int, override val type: Type, val value: String) : Lexeme(line, column) {
    override fun toString(): String = when (type) {
        Type.LINE_COMMENT -> "//$value"
        Type.BLOCK_COMMENT -> "///$value///"
        else -> throw IllegalStateException(type.string)
    }
}
